mytoyota/client.py

```diff
@@ -455,3 +456,85 @@ async def get_driving_statistics_json(
         return json.dumps(
             await self.get_driving_statistics(vin, interval, from_date), indent=3
         )
+
+    async def get_trips(self, vin: str) -> list(Trip):
+        """Returns a list of trips.
+
+        Retrieves and formats trips.
+
+        Args:
+            vin (str):
+                Vehicle identification number.
+
+        Returns:
+            A list of trips.
+
+        Raises:
+            ToyotaLoginError: An error returned when updating token or invalid login information.
+            ToyotaInternalError: An error occurred when making a request.
+            ToyotaApiError: Toyota's API returned an error.
+        """
+        _LOGGER.debug(f"Getting trips for {censor_vin(vin)}...")
+
+        raw_trips = await self.api.get_trips_endpoint(vin)
+        _LOGGER.debug(f"received {len(raw_trips.get('recentTrips',[]))} trips")
+        return raw_trips
+
+    async def get_trip(self, vin: str, trip_id: str) -> DetailedTrip:
+        """Returns a trip.
+
+        Retrieves and formats a trip.
+
+        Args:
+            vin (str):
+                Vehicle identification number.
+            trip_id (str):
+                Trip id, UUID
+
+        Returns:
+            A trip.
+
+        Raises:
+            ToyotaLoginError: An error returned when updating token or invalid login information.
+            ToyotaInternalError: An error occurred when making a request.
+            ToyotaApiError: Toyota's API returned an error.
+        """
+        trip_id = trip_id.upper()
+        _LOGGER.debug(f"Getting trip {trip_id} for {censor_vin(vin)}...")
+
+        raw_trip = await self.api.get_trip_endpoint(vin, trip_id)
+        _LOGGER.debug(f"received trip {trip_id}")
+        return raw_trip

```

Comment:
I would expect that `raw_trip` is converted to a `DetailedTrip` which should be returned, however that is not done here. May I suggested to add a constructor to `DetailedTrip`, which maintains a reference to the `raw_trip` dictionary, and provides the data through the properties of `DetailedTrip`? That newly constructed `DetailedTrip` can then be returned here.

Reply:
Yeah good point, I was originally thinking of subclassing the `VehicleData` type, but decided against it since Trips aren't technically Vehicle Data, even if the methods of that class are what I needed. Maybe a refactoring of it to something like `APIData` or `APIResponse` would make sense but I figured to leave it for another refactor PR.
I updated the commit in https://github.com/DurgNomis-drol/mytoyota/pull/188/commits/ea43d7a43bff9c20f0c3d93650efab802ed4d6d7. 

Also, I decided to create a list of `Trip`s in  [get_trips](https://github.com/DurgNomis-drol/mytoyota/pull/188/commits/ea43d7a43bff9c20f0c3d93650efab802ed4d6d7#diff-3f2665ca5babe63f37af673eed32af87e525d17c73d2a13c884f122553ffa75bR481), however that means that to get the json, I have to [unroll it in get_trips_json](https://github.com/DurgNomis-drol/mytoyota/pull/188/commits/ea43d7a43bff9c20f0c3d93650efab802ed4d6d7#diff-3f2665ca5babe63f37af673eed32af87e525d17c73d2a13c884f122553ffa75bR523). LMK if that sounds good to you

Reply:
Creating a list of `Trip` in `get_trips` is OK for me.  Let's leave it as you have implemented now.

And yes, I agree `VehicleData` is not the best name anymore now. I like your suggestion to change it to `APIResponse` in a future PR.